### PR TITLE
8.0 stable

### PR DESCRIPTION
**Pull request by Beardev118**
## What does this PR do?
 - Fixed tests with to_s comparison for issue #359

### CODE REVIEW

**Review by keithdoggett — COMMENTED** [review]
Thanks for submitting this @Beardev118. Like I mentioned in my comment the goal is to be able to compare objects directly and not attributes since that used to work, but isn't anymore. A good first step would be identifying which attributes are different from each other and we can then work towards identifying the issue.

File: test/cases/basic_test.rb [hunk 1]
```diff
@@ -106,7 +106,8 @@ def test_set_point_wkt_wrong_type
     def test_default_value
       create_model
       obj = SpatialModel.create
-      assert_equal factory.point(0, 0).to_s, obj.default_latlon.to_s
+      assert_equal factory.point(0, 0).x, obj.default_latlon.x
+      assert_equal factory.point(0, 0).y, obj.default_latlon.y
```
[the remark this answers is not in the document]
Comment (Beardev118) [reply test/cases/basic_test.rb:110]: Fixed this issue

**Review by keithdoggett — CHANGES_REQUESTED** [review]
Good catch thanks for looking into that. We can now apply that to the other failing test too,

File: test/cases/basic_test.rb [hunk 2]
```diff
@@ -122,7 +122,9 @@ def test_custom_factory
       object.area = area
       object.save!
       object.reload
-      assert_equal area.to_s, object.area.to_s
+      area_p = MyPolygon.new(area)
+      obj_area_p = MyPolygon.new(object.area)
+      assert_equal area_p, obj_area_p
```
[the remark this answers is not in the document]
Comment (Beardev118) [reply test/cases/basic_test.rb]: Updated

**Comment by Beardev118** [comment]
Hello, Keith
Would you please tell me how can I test the failed cases in the local?
With current test code, I can't find the failed cases in the local

**Comment by keithdoggett** [comment]
It's only failing on postgres 11. I assume it has something to do with not being able to cast the WKT for the default value to the proper SRID now that it's been changed to 3857.

Thinking about this more I think the solution is to change the SRID of that column back to 0 and instead modify the `factory` method in `test_helper` to accept an srid parameter.

```ruby
def factory(srid: 3857)
      RGeo::Cartesian.preferred_factory(srid: srid)
end
```
And when we're comparing data to the `default_latlon` we do

```ruby
      assert_equal factory(srid: 0).point(0, 0), obj.default_latlon
```

**Comment by Beardev118** [comment]
All workflow passed

image

**Comment by keithdoggett** [comment]
Thanks @Beardev118 this looks good. Last question is any reason this is pointing to 8.0-stable rather than main?

**Comment by Beardev118** [comment]
Thank you for your response
You wrote: "In the update to version 8, a few tests have to use to_s to
compare result geometries instead of the objects directly. We should be
able to compare the objects directly, though."
So, I started with 8.0-stable branch, but it can be applied  to the main
branch
If you want, I'll edit it to the main branch

**Review by Beardev118 — COMMENTED** [review]
Updated

**Comment by keithdoggett** [comment]
Thank you! LGTM merging

**Comment by Beardev118** [comment]
You're welcome!

**Pull request merged**